controllers/spacebindingrequest/spacebindingrequest_controller.go

```diff
@@ -92,7 +99,360 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
+func (r *Reconciler) addFinalizer(logger logr.Logger, memberCluster cluster.Cluster, spaceBindingRequest *toolchainv1alpha1.SpaceBindingRequest) error {
+	// Add the finalizer if it is not present
+	if !util.HasFinalizer(spaceBindingRequest, toolchainv1alpha1.FinalizerName) {
+		logger.Info("adding finalizer on SpaceBindingRequest")
+		util.AddFinalizer(spaceBindingRequest, toolchainv1alpha1.FinalizerName)
+		if err := memberCluster.Client.Update(context.TODO(), spaceBindingRequest); err != nil {
+			return errs.Wrap(err, "error while adding finalizer")
+		}
+	}
+	return nil
+}
+
+func (r *Reconciler) ensureSpaceBinding(logger logr.Logger, memberCluster cluster.Cluster, spaceBindingRequest *toolchainv1alpha1.SpaceBindingRequest) (bool, error) {
+	logger.Info("ensuring spacebinding")
+
+	// find space from namespace labels
+	space, err := r.getSpace(memberCluster, spaceBindingRequest)
+	if err != nil {
+		return false, r.setStatusFailedToCreateSpaceBinding(logger, memberCluster, spaceBindingRequest, err)
+	}
+	// space is being deleted
+	if util.IsBeingDeleted(space) {
+		return false, errs.New("space is being deleted")
+	}
+
+	// validate MUR
+	mur, err := r.getMUR(spaceBindingRequest)
+	if err != nil {
+		return false, r.setStatusFailedToCreateSpaceBinding(logger, memberCluster, spaceBindingRequest, err)
+	}
+	// mur is being deleted
+	if util.IsBeingDeleted(mur) {
+		return false, errs.New("mur is being deleted")
+	}
+
+	// validate Role
+	if err := r.validateRole(spaceBindingRequest, space); err != nil {
+		return false, r.setStatusFailedToCreateSpaceBinding(logger, memberCluster, spaceBindingRequest, err)
+	}
+
+	// create spacebinding if not found for given spaceBindingRequest
+	spaceBinding, err := r.getSpaceBinding(spaceBindingRequest)
+	if err != nil {
+		return false, err
+	}
+	// spacebinding not found, creating it
+	if spaceBinding == nil {
+		if err = r.createNewSpaceBinding(logger, memberCluster, spaceBindingRequest, mur, space); err != nil {
+			// failed to create spacebinding
+			return false, r.setStatusFailedToCreateSpaceBinding(logger, memberCluster, spaceBindingRequest, err)
+		}
+		return true, nil
+	}
+
+	logger.Info("SpaceBinding already exists")
+	return r.updateExistingSpaceBinding(logger, spaceBindingRequest, spaceBinding)
+}
+
+// updateExistingSpaceBinding updates the spacebinding with the config from the spaceBindingRequest.
+// returns true/nil if the spacebinding was updated
+// returns false/nil if the spacebinding was already up-to-date
+// returns false/err if something went wrong or the spacebinding is being deleted
+func (r *Reconciler) updateExistingSpaceBinding(logger logr.Logger, spaceBindingRequest *toolchainv1alpha1.SpaceBindingRequest, spaceBinding *toolchainv1alpha1.SpaceBinding) (bool, error) {
+	// check if spacebinding is being deleted
+	if util.IsBeingDeleted(spaceBinding) {
+		return false, fmt.Errorf("cannot update SpaceBinding because it is currently being deleted")
```

Comment:
Done in https://github.com/codeready-toolchain/host-operator/pull/848/commits/0d077fdc7829953cfc5ce89b87be1a06b3e2c325 

I've also moved things around so that any error returned from `ensureSpaceBinding` will end up with calling `setStatusFailedToCreateSpaceBinding` passing this error.